pkg/discovery/discovery.go

```diff
@@ -100,12 +99,16 @@ func (d *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 	// Should take failover replicas into consideration
 	if len(currentCluster.peers) == int(tc.PDStsDesiredReplicas()) && tc.Spec.Cluster == nil {
 		delete(currentCluster.peers, podName)
+		pdAddresses := tc.Spec.PDAddresses
+		// we should join an existing PD cluster
 		if len(pdAddresses) != 0 {
 			return fmt.Sprintf("--join=%s", strings.Join(pdAddresses, ",")), nil
 		}
+		// we should start a PD cluster with the full domain name
```

Comment:
```suggestion
		// Initial the PD cluster with the FQDN format service record if tc.Spec.ClusterDomain is set.
```

Reply:
FQDN: https://en.wikipedia.org/wiki/Fully_qualified_domain_name
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields